### PR TITLE
feat: Add component for highlighting news (#635)

### PR DESCRIPTION
**Pull request by gorandalum**
Add an additional parameter 'highlightComponent' to the
DisapperingHeader, which may be used for showing highlighted text and/or
images.

This new component is used on the search assistant screen where a news
banner is shown. The text and link for this news banner is fetched from
the remote config in Firebase.

### CODE REVIEW

**Comment by gorandalum** [comment]
Screenshot 2020-11-26 at 16 46 04

**Comment by rikkekantega** [comment]
Screenshot_20201126-170710_atb

The newsbanner component (or wrapper) could perhaps benefit from having a background color here, so we can control text/bg contrast.😊

File: src/components/disappearing-header/index.tsx [hunk 1]
```diff
@@ -284,6 +292,13 @@ const useThemeStyles = StyleSheet.createThemeHook((theme) => ({
     right: 0,
   },
 
+  highlightComponent: {
```
Comment (rikkekantega) [inline src/components/disappearing-header/index.tsx:295]: Is there a possibility that font scaling up to 200% and/or a little more text content to the higlightComponent could possibly render off screen and become inaccessible? 
Just wanted to mention this, as we had those issues on the ticketing splash screen and onboarding views. We solved it by making the contents scrollable

**Review by rikkekantega — CHANGES_REQUESTED** [review]
This is going to be great! And whow nice to see that you're up and running, contributing to the app already! :star_struck: 
I agree on keeping images etc out of scope. Lots of value to this feature with text only for the  first edition. I left two comments to consider, though - Just recognized a couple of possible accessibility issues.

**Comment by gorandalum** [comment]
> The newsbanner component (or wrapper) could perhaps benefit from having a background color here, so we can control text/bg contrast.😊

I talked to Petter about this now. The sketches used the new colors for dark mode, which is not yet changed in the app. I'll create a new issue for changing the colors, if it doesn't already exist, and it is a requirement that the new issue is fixed before merging this issue. This was preferred over creating a temporary workaround for dark mode text in this highlight component.

File: src/components/disappearing-header/index.tsx (hunk 1 above)
Comment (gorandalum) [reply src/components/disappearing-header/index.tsx:295]: I'll look into it 👍

Comment (gorandalum) [reply src/components/disappearing-header/index.tsx:295]: Yeah, lots of text and font scaling did not look too good. Talked with Petter about this, and he was not a big fan of scrolling.

Proposed solution:
- We set a description in Firebase Remote Config that specifies the maximum length for the text field
Screenshot 2020-11-27 at 10 41 11
- A text length of up to 120 characters and link text length of up to 40 characters will fit with font size set to max on both iOS and Android, also on small screens. This is tested locally.
- Overflowing text will be hidden. This should not happen unless someone do not follow the descriptions in Remote Config.

Does this sound as a good enough solution?

**Review by rikkekantega — APPROVED** [review]
LGTM :rocket:

**Comment by gorandalum** [comment]
> > The newsbanner component (or wrapper) could perhaps benefit from having a background color here, so we can control text/bg contrast.😊
> 
> I talked to Petter about this now. The sketches used the new colors for dark mode, which is not yet changed in the app. I'll create a new issue for changing the colors, if it doesn't already exist, and it is a requirement that the new issue is fixed before merging this issue. This was preferred over creating a temporary workaround for dark mode text in this highlight component.

Change of plans. In addition to changing colors as mentions above, we also had to add support for light/dark-mode of the svg banner. Instead of starting to fix both these things we decided to change the svg banner to white and a opacity of `0.2`. This made it usable on both light and dark mode, without being to light compared to the white text in dark mode.

File: src/components/disappearing-header/index.tsx (hunk 1 above)
Comment (gorandalum) [reply src/components/disappearing-header/index.tsx:299]: After testing with the maximum accessibility text size in iPhone settings we saw that 120 characters was actually too many to be presented without overflowing. We ended up adding ScrollView, for the cases where both the number of characters and accessibility text size is maxed out. We preserve the text length descriptions in Firebase Remote Config so scrolling rarely have to be used.

**Pull request merged**